be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -914,6 +914,7 @@ void SegmentIterator::_evaluate_short_circuit_predicate(uint16_t* vec_sel_rowid_
             predicate->type() == PredicateType::GT || predicate->type() == PredicateType::GE) {
             col_ptr->convert_dict_codes_if_necessary();
         }
+        predicate->set_dict_code_if_necessary(*short_cir_column, _segment->id());
```

Comment:
Segment id is unique in a single rowset, it is not unique between different rowsets.